### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by desano**

### CODE REVIEW

**Comment by desano** [comment]
@rootical

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Node
+node_modules
+
+# Gulp
+Gulpfile.js
```
Comment (rootical) [inline .gitignore]: It should be available.

File: css/style.css [hunk 2]
```diff
@@ -0,0 +1,169 @@
+/* all styles */
```
Comment (rootical) [inline css/style.css]: There's no need to put css files to the rep. They should be ignored, while you have to provide a setup-ready repository so I could build it locally.

File: sass/_grid.scss [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    margin-left: -$gutter/2;
+    margin-right: -$gutter/2;
+  }
+  @else {
+    width: 100%;
+    max-width: $wrapper;
+    margin-left: auto;
+    margin-right: auto;
+  }
+  &:before, &:after {
```
Comment (rootical) [inline sass/_grid.scss]: Feel free to create a clearfix mixin yourself and reuse it later within the project.

File: sass/_mixins.scss [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+    color: $color;
+  }
+  &::-webkit-input-placeholder {
+    color: $color;
+    // Safari placeholder margin issue
+    text-indent: $text-indent;
+  }
+}
+
+// Appearance
```
Comment (rootical) [inline sass/_mixins.scss]: Though I am glad you copied these ones, prefer to use autoprefixr instead.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  },
+  "author": "Michal Mrugala",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/desano/front-end-test-project/issues"
+  },
+  "homepage": "https://github.com/desano/front-end-test-project#readme",
+  "devDependencies": {
+    "gulp": "^3.9.0",
+    "gulp-connect": "^2.3.1",
```
Comment (rootical) [inline package.json]: Use browser-sync instead

File: sass/_grid.scss [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  @include mq(600px) {
+    @include col(1, false);
+  }
+  @include mq(900px) {
+    @include col(1/2, false);
+  }
+}
+.content {
+  @include col;
+  @include mq(900px) {
```
Comment (rootical) [inline sass/_grid.scss]: You should create variables for your breakpoints. Consider using breakpoint-sass.

**Comment by rootical** [comment]
Where is mobile-first approach? :(

File: package.json (hunk 5 above)
Comment (desano) [inline package.json]: Wow, browser-sync works great

File: sass/_mixins.scss (hunk 4 above)
Comment (desano) [inline sass/_mixins.scss]: Ok, I change to autoprefixr

File: .gitignore (hunk 1 above)
Comment (desano) [inline .gitignore]: Yes, of course.

File: index.html [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+                <img src="" alt=""/>
+              </figure>
+              <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus sapien diam, interdum sed consectetur a, congue vel turpis. Nam commodo erat vulputate imperdiet varius. Donec consequat mauris mollis eros cursus, sed sagittis augue convallis. Proin feugiat fermentum tincidunt. </p>
+            </div>
+          </div>
+        </section>
+      </article>
+    </div>
+    <aside class="sidebar">
+      <div class="">
```
Comment (rootical) [inline index.html]: What for is this div?

File: index.html [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+  <title>Xplosion</title>
+  <meta name="description" content="Xplosion site">
+  <meta name="author" content="Michal Mrugala">
+  <meta content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=0" name="viewport">
+  <!-- styles -->
+  <link rel="stylesheet" href="/css/style.css">
+</head>
+<body>
+  <div class="wrapper">
+    <header class="main-header clearfix">
```
Comment (rootical) [inline index.html]: Please put roles within the document.

img

File: sass/_main.scss [hunk 9]
```diff
@@ -0,0 +1,254 @@
+/* main style */
+
+// simple reset
```
Comment (rootical) [inline sass/_main.scss:3]: Use bower/npm https://github.com/necolas/normalize.css/ instead

**Comment by desano** [comment]
@rootical look at my new changes :)

File: Gulpfile.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+var autoprefixer = require('autoprefixer');
+var buffer = require('vinyl-buffer');
+var imagemin = require('gulp-imagemin');
+var merge = require('merge-stream');
+var spritesmith = require('gulp.spritesmith');
+var browserSync = require('browser-sync').create();
+
+// sass, postcss, autoprefixer task
+gulp.task('styles', function() {
+  gulp.src('sass/**/*.scss')
```
Comment (rootical) [inline Gulpfile.js:14]: Every gulp task should return. Here and below. And I told you what it could be :)

File: public/index.html [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+  <meta name="description" content="Xplosion site">
+  <meta name="author" content="Michal Mrugala">
+  <meta content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=0" name="viewport">
+  <!-- styles -->
+  <link rel="stylesheet" href="/css/style.css">
+</head>
+<body>
+  <div class="wrapper">
+    <!-- header -->
+    <header class="main-header clearfix">
```
Comment (rootical) [inline public/index.html:15]: Please put role attributes everywhere.

File: sass/_main.scss [hunk 12]
```diff
@@ -0,0 +1,467 @@
...
+    .search {
+      position: absolute;
+      top: 70px;
+      width: 100%;
+      z-index: 10;
+      left: 0;
+      background-color: #e61052;
+      background-image: linear-gradient(to top, #e61052 0%, #e61052 50%, #fb2e6d 100%);
+      white-space: nowrap;
+      max-height: 0;
```
Comment (rootical) [inline sass/_main.scss]: Why is it needed?

File: sass/_main.scss [hunk 13]
```diff
@@ -0,0 +1,467 @@
...
+      overflow: hidden;
```
Comment (rootical) [inline sass/_main.scss]: Is overflow needed?

**Comment by rootical** [comment]
Good job!
Please merge after fixing comments.

**Comment by desano** [comment]
@rootical I added last part of code & fixes to review.

**Pull request merged**